### PR TITLE
Add Container.align and Container.position

### LINKED ISSUE #4950: Container should have align and position
`align` should put an Align widget inside the decoration to align the child and `position` should put an Align widget outside the decoration to position the container within its parent.

### PR DESCRIPTION
**Pull request by abarth**
These let you add Align widget to the inside or outside of a container.
Several customers have asked for these properties.

Fixes #4950

### CODE REVIEW

**Comment by abarth** [comment]
@HansMuller @Hixie

**Comment by HansMuller** [comment]
Adding the align Container property seems like a big win, all of the examples read more easily for it.

I'm a little confused about the position parameter. It doesn't seem to have come up in the existing code (maybe I missed one?)  and I'm not sure exactly how it's to be used. Maybe add some additional explanation to the docs?

LGTM

File: packages/flutter/lib/src/widgets/container.dart [hunk 1]
```diff
@@ -114,6 +122,9 @@ class Container extends StatelessWidget {
   /// Empty space to surround the decoration.
   final EdgeInsets margin;
 
+  /// Position the container within its parent.
```
Comment (HansMuller) [inline packages/flutter/lib/src/widgets/container.dart]: Position this container within its parent.

**Comment by HansMuller** [comment]
There's certainly no harm in skipping it until a use case crops up :-)

**Comment by abarth** [comment]
Done.  PTAL.

**Comment by HansMuller** [comment]
This is pretty slick. LGTM

**Pull request merged**